tasks/release/ReleaseVersionController.php

```diff
@@ -36,15 +36,50 @@ function assignVersionToCompletedTickets($version = null) {
     return [$version, join("\n", $output)];
   }
 
-  private function getDoneIssuesWithoutVersion() {
-    $jql = "project = $this->project AND status = Done AND fixVersion = EMPTY AND updated >= -52w";
-    $result = $this->jira->search($jql, ['key']);
-    return array_map(function ($issue) {
-      return [
-        'id' => $issue['id'],
-        'key' => $issue['key'],
-      ];
-    }, $result['issues']);
+  function determineNextVersion() {
+    $last_version = $this->jira->getLastReleasedVersion();
+    $parsed_last_version = VersionHelper::parseVersion($last_version['name']);
+
+    $version_to_increment = VersionHelper::PATCH;
+
+    if ($this->project === JiraController::PROJECT_MAILPOET) {
+      $free_increment = $this->checkProjectVersionIncrement(JiraController::PROJECT_MAILPOET);
+      $premium_increment = $this->checkProjectVersionIncrement(JiraController::PROJECT_PREMIUM);
+
+      if (in_array(VersionHelper::MINOR, [$free_increment, $premium_increment])) {
+        $version_to_increment = VersionHelper::MINOR;
+      }
+    }
+
+    $parsed_last_version[$version_to_increment]++;
+    $next_version = VersionHelper::buildVersion($parsed_last_version);
+    return $next_version;
+  }
+
+  private function checkProjectVersionIncrement($project) {
+    $this->jira->setProject($project);
+    $issues = $this->getUnreleasedDoneIssues();
+    $this->jira->setProject(JiraController::PROJECT_MAILPOET); // restore project
```

Comment:
This doesn't restore the original value but always sets project "MAILPOET" instead I think.

What I'd suggest here is to get rid of the `setProject` setter on `JiraController` and pass it as a parameter to the exact method that needs it. It seems to be `getUnreleasedDoneIssues` below but actually that one seems to have project already in the JQL query 🤔

Reply:
Nice find. Fixed as you suggested.